File: go-dtm-client/GoDtmUi/Models/Session.cs
using Newtonsoft.Json;



namespace GoDtmUI.Models
{
    public class Session
    {
        [JsonProperty("token")]
        public string Token { get; set; }

        [JsonProperty("user")]
        public User User { get; set; }

        [JsonProperty("expired")]
        public int Expired { get; set; }


        public static Session FromJson(string json) => JsonConvert.DeserializeObject<Session>(json, Converter.Settings);
    }
}

File: go-dtm-server/models/session.go
package models

import (
	"time"

	uuid "github.com/satori/go.uuid"
)

const (
	TOKEN_DEFAULT_TTL = 86400
)

type Session struct {
	Token       string `json:"token"`
	User        *User  `json:"user"`
	ExpiredTime int64  `json:"expired"`
}

func (this *Session) IsAlive() bool {
	return this.ExpiredTime > time.Now().Unix()
}

func NewSession(user *User) *Session {
	v4 := uuid.NewV4()
	token := v4.String()

	session := &Session{
		Token:       token,
		User:        user,
		ExpiredTime: time.Now().Unix() + TOKEN_DEFAULT_TTL,
	}

	return session
}
